src/Components/DonationFormModal.js

```diff
@@ -11,11 +50,85 @@ const DonationFormModal = () => {
     type: '',
     gender: '',
     petDescription: '',
-    imageUpload: ''
+    imageName: '',
+    imageUrl: ''
   })
+  const [createPet] = useMutation(CREATE_PET, {
```

Comment:
Thank you for the research you did to find out how to do the immediate refetch, it was super helpful and helped with optimizing the site.